collate test fails

The collate test is currently failing:

```
$ npm test                                                                                                                                                           (master) 

> jsfbp@0.1.0 test /Users/danrumney/Projects/jsfbp
> mocha --ui mocha-fibers --recursive -R spec --require test/test_helper.js



  collate
Start time: 5/25/2016, 10:55:24 AM
Elapsed time in millisecs: 8
    1) should collate based on a single field

  copier
Start time: 5/25/2016, 10:55:24 AM
Elapsed time in millisecs: 0
    ✓ should copy multiple IPs

  delay
Start time: 5/25/2016, 10:55:24 AM
Elapsed time in millisecs: 1003
    ✓ should exactly delay a single IP (1005ms)
Start time: 5/25/2016, 10:55:25 AM
Elapsed time in millisecs: 3040
    ✓ should exactly delay multiple IPs (3040ms)

  randdelay
Start time: 5/25/2016, 10:55:29 AM
Elapsed time in millisecs: 106
    ✓ should randomly delay a single IP (108ms)
Start time: 5/25/2016, 10:55:29 AM
Elapsed time in millisecs: 3715
    ✓ should randomly delay multiple IPs (3717ms)

  FiberRuntime
Start time: 5/25/2016, 10:55:32 AM
Elapsed time in millisecs: 0
    ✓ should be able to run a simple sender-->receiver setup

  utils
    ✓ should create an enum and provide a lookup function
    ✓ should not be writable


  8 passing (8s)
  1 failing

  1) collate should collate based on a single field:

      Uncaught AssertionError: expected [ Array(18) ] to deeply equal [ Array(10) ]
      + expected - actual

       [
      -  [null]
         "1,m1"
         "1,d11"
         "1,d12"
      -  [null]
      -  [null]
         "2,m2"
         "2,d21"
      -  [null]
      -  [null]
         "3,m3"
         "3,d31"
         "3,d32"
         "3,d33"
      -  [null]
      -  [null]
         "4,d41"
      -  [null]
       ]
      
      at Assertion.assertEqual (node_modules/chai/lib/chai/core/assertions.js:485:19)
      at Assertion.ctx.(anonymous function) [as equal] (node_modules/chai/lib/chai/utils/addMethod.js:41:25)
      at Context.<anonymous> (test/components/collate.js:24:28)
      at node_modules/mocha-fibers/lib/mocha-fibers.js:24:14



npm ERR! Test failed.  See above for more details.
[danrumney:~/Projects/jsfbp]$                         
```

It looks like the `null` IPs are being propagated instead of dropped.

--- a/test/test_helper.js
+++ b/test/test_helper.js
@@ -1,6 +1,7 @@
 'use strict';
 
 var chai = require('chai');
+var IP = require('../core/IP');
 
 global.expect = chai.expect;
 
@@ -11,16 +12,18 @@ global.MockSender = function(inputArray) {
     inputArray.forEach(function(item) {
       outport.send(proc.createIP(item));
     });
-  }
-}
+  };
+};
 
 global.MockReceiver = function(outputArray) {
   return function() {
     var inport = this.openInputPort('IN');
     var ip;
     while ((ip = inport.receive()) !== null) {
-      outputArray.push(ip.contents);
+      if(ip.type === IP.NORMAL) {
+        outputArray.push(ip.contents);
+      }
       this.dropIP(ip);
     }
-  }
-}
+  };
+};

